api/v1/verticadb_webhook.go

```diff
@@ -1765,6 +1769,85 @@ func (v *VerticaDB) checkNewSBoxOrSClusterShutdownUnset(allErrs field.ErrorList)
 	return allErrs
 }
 
+// checkShutdownForScaleUpOrDown ensures a subcluster to be scaled up/down has Shutdown field set to false and
+// live in a sandbox where Shutdown is set to false in spec/status for all pieces
+func (v *VerticaDB) checkShutdownForScaleUpOrDown(oldObj *VerticaDB, allErrs field.ErrorList) field.ErrorList {
+	newSclusterMap := v.GenSubclusterMap()
+	newSclusterSboxMap := v.GenSubclusterSandboxMap()
+	newSandboxMap := v.GenSandboxMap()
+	statusSclusterIndexMap := v.GenStatusSClusterIndexMap()
+	oldSclusterMap := oldObj.GenSubclusterMap()
+	newSclusterIndexMap := v.GenSubclusterIndexMap()
+	for subclusterName, sandboxName := range newSclusterSboxMap {
```

Comment:
This loop will only check subclusters in the sandbox. What if the subclusters not in a sandbox?

Reply:
Isn't this function for this rule: you cannot scale (up/down) a subcluster with shutdown true(spec or status)?

Reply:
This will not include subclusters in the main cluster. You should iterate over `new.spec.subclusters`, for each subcluster , check if size has changed, if true, check if shutdown(spec/status) is true, if true append an error, then check if it is part of a sandbox, if true check sandbox.Shutdown, if true, append an error

Reply:
Rectification, given that there is already a rule that prevents scaling a subcluster that is part of a sandbox, you  do not need to check the sandbox.